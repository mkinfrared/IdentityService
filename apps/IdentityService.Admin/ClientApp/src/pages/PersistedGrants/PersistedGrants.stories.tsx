import { Meta, StoryObj } from "@storybook/react";

import PersistedGrants from "./PersistedGrants";
import css from "./Story.module.scss";

type Story = StoryObj<typeof meta>;

const meta = {
  title: "UI/PersistedGrants",
  component: PersistedGrants,
  tags: ["autodocs"],
} satisfies Meta<typeof PersistedGrants>;

const Default: Story = {
  args: {},
  render: (args) => (
    <div className={css.Story}>
      <PersistedGrants {...args} />
    </div>
  ),
};

export { Default };

export default meta;
