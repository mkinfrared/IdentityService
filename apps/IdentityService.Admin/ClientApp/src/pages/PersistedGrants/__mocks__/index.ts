export { default } from "./PersistedGrants.mock";
