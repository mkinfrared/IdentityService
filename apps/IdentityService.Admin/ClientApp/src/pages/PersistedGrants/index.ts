export { default } from "./PersistedGrants";

export * from "./PersistedGrants.type";
