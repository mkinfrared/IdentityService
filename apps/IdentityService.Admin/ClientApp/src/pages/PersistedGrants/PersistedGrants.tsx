import css from "./PersistedGrants.module.scss";
import { PersistedGrantsProps } from "./PersistedGrants.type";

const PersistedGrants = ({ className }: PersistedGrantsProps) => {
  const classNames = [css.PersistedGrants, className].filter(Boolean).join(" ");

  return (
    <div className={classNames} data-testid="PersistedGrants">
      PersistedGrants works!!
    </div>
  );
};

export default PersistedGrants;
