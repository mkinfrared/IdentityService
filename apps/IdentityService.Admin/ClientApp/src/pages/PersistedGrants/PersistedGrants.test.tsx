import { render } from "@testing-library/react";

import PersistedGrants from "./PersistedGrants";

describe("<PersistedGrants />", () => {
  const Component = <PersistedGrants />;

  it("should be defined", () => {
    expect(PersistedGrants).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("PersistedGrants");

    expect(element).toBeDefined();
  });
});
