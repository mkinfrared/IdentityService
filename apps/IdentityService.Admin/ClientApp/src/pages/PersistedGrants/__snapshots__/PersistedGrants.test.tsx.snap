// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PersistedGrants /> should match the snapshot 1`] = `
<div>
  <div
    class="PersistedGrants"
    data-testid="PersistedGrants"
  >
    PersistedGrants works!!
  </div>
</div>
`;
